infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/CommonPVCStrategy.java

```diff
@@ -242,8 +251,33 @@ public void cleanup(Workspace workspace) throws InfrastructureException {
     AccountImpl account = ((WorkspaceImpl) workspace).getAccount();
     if (isPersonalAccount(account) && accountHasNoWorkspaces(account)) {
       log.debug("Deleting the common PVC: '{}',", configuredPVCName);
-      deleteCommonPVC(workspace);
-      return;
+      String namespace = infraNamespace.replaceFirst("<username>", workspace.getNamespace());
+
+      // wait until all running rm pods have finished before deleting common PVC
+      try (KubernetesClient client = new DefaultKubernetesClient()) {
+        PodList pods = client.pods().inNamespace(namespace).list();
+        for (Pod pod : pods.getItems()) {
+          String podName = pod.getMetadata().getName();
+          if (podName.startsWith(PVCSubPathHelper.RM_COMMAND_BASE[0])) {
+            client
+                .pods()
+                .inNamespace(namespace)
+                .withName(podName)
+                .waitUntilCondition(CommonPVCStrategy::podRemoveCondition, 5, TimeUnit.MINUTES);
+          }
+        }
+
+      } catch (KubernetesClientException e) {
+        e.printStackTrace();
+      }
+
+      // check if account has no workspaces again, since
+      // a new workspace could have started while waiting for
+      // rm pods to finish
+      if (accountHasNoWorkspaces(account)) {
+        deleteCommonPVC(workspace);
```

Comment:
maybe here we can also double-check rm pods  and do terminate them before the PVC removal?